### PR TITLE
docs: HOCON spec compliance checklist (S1-S26, 209 items)

### PR DESCRIPTION
**Pull request by y1o1**
## Summary

Adds the per-implementation HOCON spec compliance tracker for go.hocon under [`docs/spec-compliance.md`](docs/spec-compliance.md). Inherits the 209 item IDs from the canonical template in [o3co/xx.hocon](https://github.com/o3co/xx.hocon/blob/main/docs/spec-checklist.md) and adds go.hocon-specific `tests:` and `status:` cells.

- Item count: **209**
- Status distribution: ✅ 109 / ⚠️ 2 / ❌ 6 / 🤷 72 / ➖ 20
- Compliance rate: **spec-total 52.6%** / **in-scope 58.2%**

Notable findings surfaced by the checklist (verified by Codex round-2 review):

- ⚠️ **S10.13** (array/object in string concat is an error): `TestResolver_ArrayConcatenationPermissive` at `internal/resolver/resolver_test.go:567` asserts `a = [1, 2] 3` → `[1, 2, 3]` (array followed by scalar in concat); spec L373 says arrays/objects in string concat must error. Permissive extension pinned by test.
- ⚠️ **S13.11** ([#45](https://github.com/o3co/go.hocon/issues/45) still OPEN): lenient mode drops optional substitutions in nested-include scope.
- ❌ **S11.4** (path expression `10.0foo` → `[10, 0foo]`): parser keys accept `TokenString` / `TokenInt` but not `TokenFloat` (`internal/parser/parser.go:248`), so `10.0foo = x` is rejected rather than producing the spec-defined path split.
- ❌ **S13c.1–S13c.5** (`\${X[]}` env-var list): `internal/lexer/lexer.go:397` rejects `[` / `]` inside `\${...}` body. Feature unparseable.
- 72 🤷 items remain as test debt, concentrated in S5 comma error cases, S6 Unicode whitespace details, S8 unquoted-string edge cases, S10 concat error cases, S13/S13a substitution edge cases, S15 numeric-indexed object → array (entire section uncovered), S17 type-conversion error cases.

20 items inherited from the template as globally out-of-scope: classpath (JVM-only), URL include (READMEs declare unsupported), Period Format (mirrors `java.time.Period`), reference.conf / application.conf (JVM convention), MIME type, surrogate codepoint (intentional language-natural divergence), `.properties` multi-line/Unicode escape simplification.

The companion cross-implementation roll-up lives at [xx.hocon/docs/compliance-matrix.md](https://github.com/o3co/xx.hocon/blob/main/docs/compliance-matrix.md).

## Test plan

- [x] Item count: 209 (matches template)
- [x] ID order: exact match with template (`diff` empty)
- [x] All ➖ items inherit the template's `out-of-scope:` rationale verbatim
- [x] `go test ./...` all green (220 passed, 2 intentionally skipped)
- [x] Codex review round 2 applied (S3.2, S10.2/13, S11.4, S13.10, S13a.12/14, S13c.1-5, S20)
- [ ] Reviewer spot-check on a section of choice
- [ ] Decide whether to file follow-up issues for the test-debt clusters

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: docs/spec-compliance.md [hunk 1]
```diff
@@ -0,0 +1,937 @@
+# HOCON Spec Compliance — go.hocon
+
+This file extends [`xx.hocon/docs/spec-checklist.md`](../../xx.hocon/docs/spec-checklist.md) with go.hocon-specific status and test references. It inherits all 209 items in the same order; description lines are verbatim from the template and must not be edited here.
```
[the remark this answers is not in the document]
Comment (y1o1) [reply docs/spec-compliance.md]: Fixed in 48468cd — replaced the relative path with an absolute `github.com/o3co/xx.hocon` URL.
